cdap-app-fabric/src/main/java/io/cdap/cdap/internal/capability/autoinstall/HubPackage.java

```diff
@@ -0,0 +1,254 @@
+
+  private final String name;
+  private final String version;
+  private final String description;
+  private final String label;
+  private final String author;
+  private final String org;
+  private final String cdapVersion;
+  private final long created;
+  private final boolean beta;
+  private final List<String> categories;
+  private final boolean paid;
+  private final String paidLink;
+  private transient ArtifactVersionRange versionRange;
+  private transient ArtifactVersion artifactVersion;
+
+  public HubPackage(String name, String version, String description, String label, String author, String org,
+                    String cdapVersion, long created, boolean beta, List<String> categories,
+                    boolean paid, String paidLink) throws InvalidArtifactRangeException {
+    this.name = name;
+    this.version = version;
+    this.artifactVersion = new ArtifactVersion(version);
+    this.description = description;
+    this.label = label;
+    this.author = author;
+    this.org = org;
+    this.cdapVersion = cdapVersion;
+    this.versionRange = ArtifactVersionRange.parse(cdapVersion);
+    this.created = created;
+    this.beta = beta;
+    this.categories = categories;
+    this.paid = paid;
+    this.paidLink = paidLink;
+  }
+
+  private String getCdapVersion() {
+    return cdapVersion;
+  }
+
+  public ArtifactVersion getArtifactVersion() {
+    if (artifactVersion == null) {
+      // artifactVersion will be null if the class was deserialized from json.
+      artifactVersion = new ArtifactVersion(version);
+
+    }
+    return artifactVersion;
+  }
+
+  public boolean versionIsInRange(String version) {
+    if (versionRange == null) {
+      // versionRange will be null if the class was deserialized from json.
+      if (cdapVersion == null) {
+        return false;
+      }
+      try {
+        versionRange = ArtifactVersionRange.parse(cdapVersion);
+      } catch (InvalidArtifactRangeException e) {
+        LOG.error(String.format("%s has invalid artifact range", name), e);
+        return false;
+      }
+    }
+    return versionRange.versionIsInRange(new ArtifactVersion(version));
+  }
+  public void installPlugin(URL url, ArtifactRepository artifactRepository, File tmpDir) throws Exception {
+    // Deserialize spec.json
+    URL specURL = new URL(url, "/v2/packages/" + name + "/" + version + "/spec.json");
+    Spec spec = GSON.fromJson(HttpClients.doGetAsString(specURL), Spec.class);
+    for (Spec.Action action: spec.getActions()) {
+      // Get plugin jar and json names under one_step_deploy_plugin action in the spec
+      // See https://cdap.atlassian.net/wiki/spaces/DOCS/pages/554401840/Hub+API?src=search#one_step_deploy_plugin
+      if (!action.getType().equals("one_step_deploy_plugin")) {
+        continue;
+      }
+
+      String configFilename = action.getConfigFilename();
+      if (configFilename == null) {
+        LOG.error("Config not found for plugin {}", name);
```

Comment:
Use LOG.warn if it doesn't actually causing error (e.g. in this case, it just get ignored). Also, state clearly in the log message that it is getting ignored (e.g. "Ignoring plugin {} due to missing config")

Reply:
fixed